common/user-entity-data.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "components/transforms.hpp"
+
+namespace tec {
+namespace user {
+struct EntityData {
+	Position position;
+	Orientation orientation;
+
+	// Called from User::RegisterLuaType
+	static void RegisterLuaType(sol::state& state) {
+		// clang-format off
+		state.new_usertype<EntityData>(
+			"EntityData", sol::no_constructor,
+			"position", &EntityData::position,
+			"orientation", &EntityData::orientation
+		);
+		// clang-format on
```

Comment:
The duplicate file is because visual studio did me dirty when I was moving files and saved a second copy. Yes as time progresses there will be more entity states that should be saved/loaded. I just picked 2 that made the most sense for this moment.